hikari/impl/entity_factory.py

```diff
@@ -402,15 +410,19 @@ def deserialize_audit_log(self, payload: data_binding.JSONObject) -> audit_log_m
             changes = []
             if (change_payloads := entry_payload.get("changes")) is not None:
                 for change_payload in change_payloads:
-                    key: typing.Union[audit_log_models.AuditLogChangeKey, str]
-                    key = audit_log_models.AuditLogChangeKey(change_payload["key"])
+                    key: typing.Union[audit_log_models.AuditLogChangeKey, str] = audit_log_models.AuditLogChangeKey(
+                        change_payload["key"]
+                    )
 
                     new_value: typing.Any = change_payload.get("new_value")
                     old_value: typing.Any = change_payload.get("old_value")
                     if value_converter := self._audit_log_entry_converters.get(key):
                         new_value = value_converter(new_value) if new_value is not None else None
                         old_value = value_converter(old_value) if old_value is not None else None
 
+                    elif __debug__ and not isinstance(key, audit_log_models.AuditLogChangeKey):
+                        _LOGGER.debug("Unknown audit log change key found %r", key)
+
                     changes.append(audit_log_models.AuditLogChange(key=key, new_value=new_value, old_value=old_value))
```

Comment:
uh... doesn't this break type safety? `key` is not guaranteed to be a `AuditLogChangeKey`... (Or at least, you say it might be by putting that `if` statement above. Not sure of the context here).

Reply:
key is typed as being a Union of AuditLogChangeKey or str on AuditLogChange inline with how we type other attributes which are typed as enum values